app/controllers/reactions_controller.rb

```diff
@@ -52,7 +52,17 @@ def create
 
     Rails.cache.delete "count_for_reactable-#{params[:reactable_type]}-#{params[:reactable_id]}"
 
+    if params[:reactable_type] == "Article"
```

Comment:
I feel like this whole action has gotten uncomfortably big. Since you're here already, do you maybe want to see if we can refactor things a bit?

Reply:
I agree enthusiastically but I'm wary of doing it alone 😅

Reply:
Okay I abstracted the code that handles an existing reaction - hope this works?

Reply:
@msarit just trust yourself 😃